starfish/image/_stack.py

```diff
@@ -83,9 +84,22 @@ def __init__(self, image_partition):
             if ix == axis_data.order
         ]
         shape.extend(self._tile_shape)
+        dims = [
```

Comment:
I find nested for loops in list comprehensions a bit confusing. I can never remember which way to read them. What are your thoughts on this vs creating the empty list and appending to it? 

Second, do we still need AXES_DATA with this change to ImageStack? It buys us optionality, right? Can we delegate access to the information in AXES_DATA to the xarray.dims attribute?

Reply:
> I find nested for loops in list comprehensions a bit confusing. I can never remember which way to read them. What are your thoughts on this vs creating the empty list and appending to it?

i don't think it's confusing, but i can switch it.  rule for reading: for-loops from top to bottom.

> Second, do we still need AXES_DATA with this change to ImageStack? It buys us optionality, right? Can we delegate access to the information in AXES_DATA to the xarray.dims attribute?

yes, we still need it, because we still build the array in that order.  we can make it hidden if you think it is beneficial. (but in separate PR)